doc/devguide/codebase/code-style.rst

```diff
@@ -6,12 +6,165 @@ Suricata uses a fairly strict coding style. This document describes it.
 Formatting
 ~~~~~~~~~~
 
+clang-format
+^^^^^^^^^^^^
+
+.. Argh, github does not support admonitions such as .. note::
+
++-------------------------------------------------------+
+| **Note**                                              |
+|                                                       |
+| The ``.clang-format`` file requires clang 9 or newer. |
++-------------------------------------------------------+
+
+``clang-format`` is configured to help you with formatting C code.
+
+
+Use ``git clang-format`` to only format the changes in your branch.
+
+Format your Changes
+*******************
+Before opening a pull request, please also try to ensure it is formatted
+properly. We use ``clang-format`` for this, which has git integration through the
+``git-clang-format`` script. On some systems, it may already be installed (or
+be installable via your package manager). If so, you can simply run it – the
+following command will format only the code changed in the most recent commit:
+
+.. code-block:: bash
+
+    $ git clang-format HEAD~1 # or HEAD^
+
+Note that this modifies the files, but doesn’t commit them – you’ll likely want to run
+
+.. code-block:: bash
+
+    $ git commit --amend -a
+
+in order to update the last commit with all pending changes.
+
+Use make targets
+""""""""""""""""
+Make targets are provided for easier usage, especially if you have multiple
+commits to format.
+
+Check if you branch changes' formatting is correct with:
+
+.. code-block:: bash
+
+    $ make check-style-branch
+
+Format changes in git staging and unstaged changes, i.e. code you are currently
+editing.
+
+.. code-block:: bash
+
+    $ make style
+    # Or with git clang-format
+    $ git clang-format --force
+
+Format every commits in your branch off master and rewrite history using the
```

Comment:
Format every commits -> Format every commit

Man, my spelling sucks. And as soon as one opens a pull requests, they pop up ;)

Reply:
will be fixed in v5